packages/babel-plugin-proposal-export-default-from/src/index.js

```diff
@@ -33,8 +33,12 @@ export default declare(api => {
           nodes.push(node);
         }
 
-        const [importDeclaration] = path.replaceWithMultiple(nodes);
-        path.scope.registerDeclaration(importDeclaration);
+        const [
+          importDeclaration,
+          exportNamedDeclaration,
+        ] = path.replaceWithMultiple(nodes);
+        scope.registerDeclaration(importDeclaration);
+        scope.getBinding(uid.name).reference(exportNamedDeclaration);
```

Comment:
The impact of this change might be greater than initially imagined, as now any injected import binding without updating `referencePaths` (though they should have been updated) will eventually be removed by `transform-typescript`.

Reply:
I think this is fine; this change will land in a major release, and it will give plugin authors the chance to fix their scope handling.